src/NHibernate/Loader/Criteria/CriteriaQueryTranslator.cs

```diff
@@ -863,6 +863,20 @@ public string GetEntityName(ICriteria subcriteria, string propertyName)
 			return GetEntityName(subcriteria);
 		}
 
+		internal IDictionary<string, string> GetCriteriaSQLAliasMap()
+		{
+			var result = criteriaSQLAliasMap.Where(p => !string.IsNullOrEmpty(p.Key.Alias)).ToDictionary(p => p.Key.Alias, p => p.Value);
+			if (outerQueryTranslator != null)
+			{
+				foreach (var p in outerQueryTranslator.GetCriteriaSQLAliasMap())
+				{
+					result.Add(p.Key, p.Value);
```

Comment:
Possible breaking change. At least in SQL, it is supported to have a subquery alias hiding an outer query alias. If it is also supported with Criteria, then this code needs to be adjusted.
```suggestion
					if (!result.ContainsKey(p.Key))
						result.Add(p.Key, p.Value);
```

Reply:
Fixed